src/main/resources/proto/googleidentity/oauth2/client/ClientDetails.proto

```diff
@@ -0,0 +1,40 @@
+//   Copyright 2020 Google LLC
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+syntax = "proto3";
+
+package googleidentity.oauth2.client;
+
+option java_package = "com.google.googleidentity.oauth2.client";
+option java_multiple_files = true;
+
+// Metadata about an OAuth client registered with this server.
+message ClientDetails {
+
+    string client_id = 1;
+
+    string secret = 2;
+
+    // The OAuth 2.0 grant types that are permitted by this client.
+    repeated string grant_types = 3;
+
+    // Whether scopes for this client is limited
+    bool is_scoped = 4;
+
+    // The OAuth 2.0 scopes that are permitted by this client.
+    repeated string scope = 5;
+
+    // Redirect uri for authorization endpoint
+    string redirect_uri = 6;
```

Comment:
Can redirect_uri be repeated? GAL asks 3P services to whitelist several URLs (for GAL testing and production environments).